org.xaraxtreme.XaraLX.json

```diff
@@ -0,0 +1,779 @@
+                    ]
+                },
+                {
+                    "type": "shell",
+                    "commands": [
+                        "echo 'fix configure: xml2';",
+                        "find * -xtype f \\( -name 'configure' -o -name 'configure.in' -o -name 'output.0' -o -name 'output.1' \\) | sort -V | while read -r filename; do
+                            sed -r -i 's/^([[:blank:]]*XML2CONFIG=)(xml2-config)([[:blank:]]*)$/\\1\"pkg-config libxml-2.0\"\\3/' \"${filename}\";
+                            sed -r -i 's/([[:blank:]]*|^)(test \"\\$xml2version\" -lt )([0-9]+)([^0-9]|$)/\\1\\20\\4/g' \"${filename}\";
+                        done;",
+                        "grep -HERi 'XML2CONFIG=' * || :;",
+                        "grep -HERi '\"\\$xml2version\"' * || :;"
+                    ]
+                },
+                {
+                    "type": "shell",
+                    "commands": [
+                        "echo 'autogen and autoreconf';",
+                        "cp -p /usr/share/automake-*/config.{sub,guess} .;",
+                        "./autogen.sh",
+                        "pushd 'filters/SVGFilter'; autoreconf -f -i; popd;"
+                    ]
+                },
+                {
+                    "type": "shell",
+                    "commands": [
+                        "echo 'configure';",
+                        "export WXCONFIG='wx-config';
+                        export WXCONFIG_PATH=\"$( command -v \"${WXCONFIG}\" )\";
+                        export optflags='-O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -m64 -mtune=generic';
+                        export CFLAGS=\"${optflags} -fno-strict-aliasing \";
+                        export CXXFLAGS=\"${optflags} -fno-strict-aliasing -fpermissive \";
+                        export LDFLAGS=\"${LDFLAGS} -lgobject-2.0 \";
+                        configure_args=( );
+                        configure_args+=( '--enable-unicode' );
+                        configure_args+=( '--enable-xarlib' );
+                        configure_args+=( '--enable-filters' );
+                        #configure_args+=( '--disable-svnversion' );
+                        configure_args+=( \"--with-wx-config=${WXCONFIG_PATH}\" );
+                        configure_args+=( '--prefix=/app' );
+                        configure_args+=( '--exec-prefix=/app' );
+                        echo 'configure: XaraLX';
+                        for i in \"${configure_args[@]}\"; do
+                            echo \"- '${i}'\";
+                        done;
+                        ./configure \"${configure_args[@]}\";"
+                    ]
+                },
+                {
+                    "type": "shell",
+                    "commands": [
+                        "echo 'fix configure';",
+                        "hardening_flags=( );
```

Comment:
Create a patch instead.

Reply:
Impossible. The purpose of this code is to remove some hardening flags **after** executing the configure script. Actually, this is a Fedora-specific bugfix. GNOME Platform is not affected at this point, but this [may change](https://github.com/flathub/flathub/issues/353) in the future.
As you may know, Fedora >= 23 uses the [hardening flags](https://fedoraproject.org/wiki/Changes/Harden_All_Packages) by default for the whole system. It is possible to disable it for selected packages by undefining `_hardened_build` marco in the RPM SPEC file. However, the Fedora's `wx-config` has some hard-coded flags that can not be removed in this way.
And because of the *CDraw* library, *XaraLX* does not build with `-fPIE` and `-pie` flags on x86-64 arch.

```
/usr/bin/wx-config --libs
-pthread -Wl,-z,relro -specs=/usr/lib/rpm/redhat/redhat-hardened-ld  -lwx_gtk2u_richtext-2.8 -lwx_gtk2u_aui-2.8 -lwx_gtk2u_xrc-2.8 -lwx_gtk2u_qa-2.8 -lwx_gtk2u_html-2.8 -lwx_gtk2u_adv-2.8 -lwx_gtk2u_core-2.8 -lwx_baseu_xml-2.8 -lwx_baseu_net-2.8 -lwx_baseu-2.8
```

```
$ cat /usr/lib/rpm/redhat/redhat-hardened-ld
*self_spec:
+ %{!static:%{!shared:%{!r:-pie}}}

*link:
+ -z now
$ cat /usr/lib/rpm/redhat/redhat-hardened-cc1
*cc1_options:
+ %{!r:%{!fpie:%{!fPIE:%{!fpic:%{!fPIC:%{!fno-pic:-fPIE}}}}}}

```

See also: https://lists.debian.org/debian-devel/2016/05/msg00309.html
> code compiled without -fPIC and without -fPIE can be used only in non-PIE executables